### PR TITLE
Fix network selection dropdown formatting

### LINKED ISSUE #187: Nicer network naming on select a network
Instead of all caps constant style names, we should have:
- Mainnet
- Testnet on Paseo
- Localhost
- Custom

Image

### PR DESCRIPTION
**Pull request by wilwade**
## Goal

The goal of this PR is to have better naming of the network dropdown.

Closes #187 

## Discussion

By switching to an id with an enum we get much better type safety and have a pretty name!

## Checklist

- [x] PR Self-Review and Commenting
- [x] Tests updated

## Screenshots

image

### CODE REVIEW

File: src/components/SelectNetworkAndAccount.svelte [hunk 1]
```diff
@@ -66,13 +66,13 @@
 
   async function networkChanged() {
     isLoading = true;
+    isCustomNetwork = selectedNetwork?.id === NetworkType.CUSTOM;
```
Comment (wilwade) [inline src/components/SelectNetworkAndAccount.svelte:69]: By moving this before the await, it removes the lag of the custom field appearing and disappearing

File: test/e2e/page.test.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ describe('End to End Tests', () => {
 
     // Be sure to wait for all the promises to resolve before checking the result
     await waitFor(() => {
-      expect(select).toHaveTextContent('LOCALHOST: ws://127.0.0.1:9944');
+      expect(select).toHaveTextContent('Localhost: ws://127.0.0.1:9944');
```
Comment (claireolmstead) [inline test/e2e/page.test.ts:60]:
```suggestion
      expect(select).toHaveTextContent(`${NetworkType.LOCALHOST}: ws://127.0.0.1:9944`);
```

**Review by claireolmstead — APPROVED** [review]
Read code. Great improvements. Non blocking suggestion.

File: test/e2e/page.test.ts (hunk 2 above)
Comment (wilwade) [reply test/e2e/page.test.ts:60]: That wouldn't work as that's the enum (and it isn't a string enum). That said, I'm ok with it being static.

**Pull request merged**